pkg/cloudprovider/providers/aws/instances.go

```diff
@@ -34,26 +34,26 @@ import (
 // awsInstanceRegMatch represents Regex Match for AWS instance.
 var awsInstanceRegMatch = regexp.MustCompile("^i-[^/]*$")
 
-// awsInstanceID represents the ID of the instance in the AWS API, e.g. i-12345678
+// InstanceID represents the ID of the instance in the AWS API, e.g. i-12345678
 // The "traditional" format is "i-12345678"
 // A new longer format is also being introduced: "i-12345678abcdef01"
 // We should not assume anything about the length or format, though it seems
 // reasonable to assume that instances will continue to start with "i-".
-type awsInstanceID string
+type InstanceID string
 
-func (i awsInstanceID) awsString() *string {
+func (i InstanceID) awsString() *string {
 	return aws.String(string(i))
 }
 
-// kubernetesInstanceID represents the id for an instance in the kubernetes API;
+// KubernetesInstanceID represents the id for an instance in the kubernetes API;
 // the following form
 //  * aws:///<zone>/<awsInstanceId>
 //  * aws:////<awsInstanceId>
 //  * <awsInstanceId>
-type kubernetesInstanceID string
+type KubernetesInstanceID string
```

Comment:
Exported here so we can use it from the E2E framework

Reply:
This makes it sound like it's intended to be cloud-agnostic, maybe this shouldn't live in the `aws` package?

Reply:
hmm not sure about this one TBH. I don't think it's intended to be cloud agnostic (it's definitely not used by other providers) but I can follow up with @mcrute later.

Reply:
that works for me